app/services/charges/charge_models/graduated_service.rb

```diff
@@ -9,37 +9,24 @@ def ranges
         properties['graduated_ranges']&.map(&:with_indifferent_access)
       end
 
-      def compute_amount
-        ranges.reduce(0) do |result_amount, range|
-          flat_amount = BigDecimal(range[:flat_amount])
-          per_unit_amount = BigDecimal(range[:per_unit_amount])
-
-          # NOTE: Add flat amount to the total
-          result_amount += flat_amount unless units.zero?
-
-          range_units = compute_range_units(range[:from_value], range[:to_value])
-          result_amount += range_units * per_unit_amount
-
-          # NOTE: aggregation_result.aggregation is between the bounds of the current range,
-          #       we must stop the loop
-          break result_amount if range[:to_value].nil? || range[:to_value] >= units
-
-          result_amount
-        end
+      def amount_details
+        {
+          graduated_ranges: ranges.each_with_object([]) do |range, amounts|
+            amounts << Charges::AmountDetails::RangeGraduatedService.call(range:, total_units: units)
+            break amounts if range[:to_value].nil? || range[:to_value] >= units
+          end
```

Comment:
Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.